src/script/cpp_api/s_env.cpp

```diff
@@ -206,6 +199,54 @@ bool ScriptApiEnv::read_nodenames(lua_State *L, int idx, std::vector<std::string
 	return true;
 }
 
+LuaABM *ScriptApiEnv::readABM(lua_State *L, int abm_index, int id)
+{
+	if (!lua_istable(L, abm_index)) {
+		return nullptr;
```

Comment:
i would remove this check (indexing something that isn't indexable will already trigger an appropriate lua error, so we lose no "strictness"). and it looks like that's what the old code already did, so we're not breaking anything.

if we are to keep this check, please throw an informative `LuaError` here.

this function should always return a non-null `LuaABM*`, and throw a lua error if it can't. then you can also simplify the callsites. and there is no surprising no-op behavior.